### DISCUSSION TITLE
Typescript decorators in JS files

### DISCUSSION THREAD

**Discussion by akzhy** [oven-sh/bun#4403] [Q&A]
I am trying to setup bun on an existing react project and in many places we are using the `@connect` decorator for redux (using @babel/plugin-proposal-decorators). 

When I try to run it, I get the error

```
error: Unexpected @

@connect(
```

Is there a way to achieve this?

**Comment by sirenkovladd** [comment]
might related https://github.com/oven-sh/bun/issues/4122

**Reply by akzhy** [reply]
But I am not using TypeScript, all my files are `.js`. I also tried using a `tsconfig.json` with `allowJS` instead of `jsconfig.json`

**Reply by joket1999** [reply]
The same with me. I'm using `js` not `ts`.

**Reply by akzhy** [reply]
I managed to get it working by using a very hacky solution. Before calling Bun.build, recursively copy all the files and directories in your source directory to a temp directory while renaming all .js files to .tsx. Bun.build should use the temp directory as source and it should compile without any issues since you are compiling a typescript file. @joket1999
